src/storages/request_queue.js

```diff
@@ -448,12 +458,32 @@ export class RequestQueue {
      * @returns {Promise<boolean>}
      */
     async isFinished() {
+        if (this.inProgressCount() > 0 && (Date.now() - +this.lastActivity) > this.internalTimeoutMillis) {
+            const message = `The request queue seems to be stuck for ${this.internalTimeoutMillis / 1e3}s, resetting internal state.`;
+            this.log.warning(message, { inProgress: [...this.inProgress] });
```

Comment:
Would be nice to know where it got stuck here. If it was when fetching request, or handling request, marking request as handled and so on. Can we do this?

Reply:
I can save the last action that was done next to the time, but I already know it won't help us. The problem was never about the last action that was done on the queue, but a stuck request - and that one was processed much ealier, not at the very end.

Reply:
Yeah, I meant to have this tied to the stuck request, not the one currently being processed.